src/main/java/seedu/address/model/tag/ModuleTag.java

```diff
@@ -15,15 +15,47 @@ public class ModuleTag extends Tag implements Comparable<ModuleTag> {
             "NUS Modules should have 2 - 3 letter prefix, followed by 4 digits and optional 1 - 3 alphabets";
     public static final String VALIDATION_REGEX = "[A-Z]{2,4}[0-9]{4}[A-Z]{0,3}";
 
+    private String day;
+    private String startTime;
+    private String endTime;
+    private String location;
+    private String latitude;
+    private String longitude;
+
+
     /**
      * Constructs a {@code ModuleTag}.
      *
+     * @param tagName A string for tag names.
+     */
+    public ModuleTag(String[] tagName) {
+        super(tagName[0]);
+        requireNonNull(tagName);
+        checkArgument(isValidTagName(tagName[0]), MESSAGE_CONSTRAINTS);
+        if (tagName.length == 4) {
+            this.day = tagName[1];
+            this.startTime = tagName[2];
+            this.endTime = tagName[3];
+            return;
+        }
+        // probably need to sanitize arguments somewhere either have it length 1 or length 4 etc
+        this.day = null;
+        this.startTime = null;
+        this.endTime = null;
+    }
```

Comment:
Yes please sanitise arguments. I highly recommend heavily abstracting this part because I forsee a lot of repetition.